x-pack/qa/security-example-spi-extension/src/main/java/org/elasticsearch/example/ExampleSecurityExtension.java

```diff
@@ -36,7 +36,8 @@
 public class ExampleSecurityExtension implements SecurityExtension {
 
     static {
-        if (RuntimeVersionFeature.isSecurityManagerAvailable()) {
+        final boolean useEntitlements = Boolean.parseBoolean(System.getProperty("es.entitlements.enabled"));
+        if (useEntitlements == false && RuntimeVersionFeature.isSecurityManagerAvailable()) {
```

Comment:
Why do we check for entitlements here?

Reply:
If we don't, and run with `test-entitlement`/ with entitlement enabled, there will be no Security Manager -- will be null and throw an exception.